assets/src/edit-story/app/story/storyProvider.js

```diff
@@ -26,12 +26,19 @@ import useSelectElementById from './actions/useSelectElementById';
 import useAppendElementToCurrentPage from './actions/useAppendElementToCurrentPage';
 import useSetCurrentPageByIndex from './actions/useSetCurrentPageByIndex';
 import useSetPropertiesOnSelectedElements from './actions/useSetPropertiesOnSelectedElements';
+import useSavePost from './actions/useSavePost';
 
 function StoryProvider( { storyId, children } ) {
 	// Story state is stored in these three immutable variables only!
 	// Don't update 1 of these in an effect based off another base variable.
 	// Only update these directly as a response to user or api interactions.
 	const [ pages, setPages ] = useState( [] );
+	const [ title, setTitle ] = useState( '' );
```

Comment:
Would it be better to just treat these all as story's props (just like page's props, etc) and reduce number of state read/writers? Then it'd all be part of the data model, including history undo/redo. Obviously, this would not include things like `setIsSaving`, etc.

Reply:
I separated these out, as means you can be more selective about what you write to. Different component will write to state. If all the properties are one object, it may mean a redraw of components that do not need it. @barklund Thoughts ^

Reply:
Good for now. But IMHO it's a good time to huddle again w.r.t. data model and how it's updated. In some cases the state is convenient, but I think a simpler reducer could have some benefits for us. Can we have that discussion with @barklund ?

Reply:
See https://github.com/ampproject/amp-wp/pull/3871#issuecomment-565461091.

Reply:
See https://github.com/ampproject/amp-wp/pull/3871#discussion_r357696957